Updated layout and added USD price

nenadV91 opened the pull request:
# Summary

This PR improves the layout and design a bit.
Now there is a dropdown where you can toggle between `All tokens` and `Favourite tokens`
![Screenshot from 2022-06-15 14-03-24](https://user-images.githubusercontent.com/34926005/173822740-2b462f46-ae3b-4851-8f89-a539cec1afc8.png)

Also there is a USD estimate of token balance
![Screenshot from 2022-06-15 14-26-26](https://user-images.githubusercontent.com/34926005/173826585-7268c9ca-4089-4b93-8569-9b15f8accb37.png)

Updated layout
![Screenshot from 2022-06-15 14-27-23](https://user-images.githubusercontent.com/34926005/173826745-cd9e2094-8b8a-44e7-846d-80638368dfb5.png)

![Screenshot from 2022-06-15 14-28-11](https://user-images.githubusercontent.com/34926005/173826790-0078ddbe-4d7b-4fb8-96c6-3274def18d3a.png)

anxolin commented:
@nenadV91 u need to push your workmates for reviews! If you see they ignore your PR, send some reminders in our channel

cc/ @W3stside can u review?

nenadV91 commented:
@anxolin This is on hold for now anyways as we agreed on, so its not urgent for a review

anxolin commented:
Okay, I mean, we have these as low prio, but the PR is done, so I think we can get it merge. If theres minor things we can address, we can solve. 

Then we can think what is the minimun things we would include to publish this so its useful. In my eyes, we didn't put it on hold, just deprioritized because we need the account sections first, and the analytics, and this tokens can easily get out of hand with scope creep

nenadV91 commented:
@elena-zh 
For the points you mentioned
1. Seems to work now
2. Added this and also changed to separate USD value column
3. Good catch, fixed
4. Also good catch, fixed
5. I personally like orange more, but its not a problem to change this
6. Good suggestion, changed this
7. and 8. I think for now this is fine, once we get some feedback from users we can add more changes
9. Yea this would be hard to fix, unless we add background for the table section but I don't see this as a big issue

alfetopito commented on .env:
```diff
@@ -74,5 +74,5 @@ REACT_APP_MOCK=true
 # IPFS uploading
-REACT_APP_PINATA_API_KEY=
-REACT_APP_PINATA_SECRET_API_KEY=
+REACT_APP_PINATA_API_KEY=foo
+REACT_APP_PINATA_SECRET_API_KEY=bar
```
Nice try, but nope :)

I rather untrack this file than putting trash defaults.
If we do that it'll defeat the purpose of the warning

nenadV91 replied:
Good catch, I wasn't planing on committing this, I thought this file is ignored :smile:

alfetopito commented on src/custom/components/Tokens/BalanceCell.tsx:
```diff
@@ -0,0 +1,25 @@
+  }
+
+  return (
+    <BalanceValue title={balance?.toExact()} hasBalance={!!hasBalance}>
```
Nitpick: at this point we know the balance is defined

```suggestion
    <BalanceValue title={balance.toExact()} hasBalance={!!hasBalance}>
```

Also for the `title` text [there's `formatMax`](https://github.com/cowprotocol/cowswap/blob/9ee39d414b760e67f36335c5ac46c517e1106d92/src/custom/utils/format.ts#L164=)

alfetopito commented on src/custom/components/Tokens/FiatBalanceCell.tsx:
```diff
@@ -0,0 +1,41 @@
+  if (!hasBalance) return <BalanceValue hasBalance={false}>0</BalanceValue>
+
+  return (
+    <BalanceValue title={fiatValue?.toExact()} hasBalance={!!hasBalance}>
```
Same comment as previous regarding `?.` and title amount formating

nenadV91 replied:
Update to use formatMax, but here the fiatValue can be null since we have some following logic where we want to display tooltip when there is balance but no fiatBalance.

nenadV91 commented:
@elena-zh 
1. fixed
2. should be fixed 
3. fixed
4. i can see this sometimes but can't reproduce it regurarly
5. i think its fine for now, maybe Michel can help us with this
6. I'm not sure this is related to this PR, maybe we can create a separate issue for this.

elena-zh commented:
@nenadV91 , all the changes LGTM!
However, I can always reproduce 4 case in a mobile and destop views by a connected/disconnected user
![all the time](https://user-images.githubusercontent.com/70885163/178782929-344d94cb-ca3e-459f-a032-2956e0e0836c.jpg)

alfetopito commented:
Base branch is out of sync. Can you rebase||merge with the parent?

elena-zh reviewed (approved):
LGTM now!

I believe, all UI issues we discussed above will be resolved a bit later.

W3stside commented on src/custom/components/Button/index.tsx:
```diff
@@ -254,12 +254,13 @@ export function ButtonDropdownLight({
+  stroke,
   ...rest
-}: { fill: boolean; size?: string } & HTMLAttributes<HTMLDivElement>) => {
+}: { fill: string | false; size?: string; stroke: string } & HTMLAttributes<HTMLDivElement>) => {
```
if fill is false it'll always default to 'transparent' right? just set it as a string and pass 'transparent' as the default prop in the destructure

W3stside commented on src/custom/components/Tokens/TokensTable.tsx:
```diff
@@ -182,35 +186,37 @@ export default function TokenTable({
+                <MouseoverTooltip
+                  text={
+                    <Trans>
+                      Enable token for trading. This only need to be done once. Once it is enabled, you can place orders
```
where is this text coming from?

nenadV91 replied:
From the comment Anxo made https://github.com/cowprotocol/cowswap/pull/677#issuecomment-1178796972

nenadV91 commented on src/custom/components/Tokens/FiatBalanceCell.tsx:
```diff
@@ -0,0 +1,41 @@
+    isLocaleAware: true,
+  })
+
+  if (!balance && account) return <Loader />
```
The thing is that those 2 describe different situations. If balance is undefined it means its not loaded yet and we need to detect that, and hasBalance also detects if the balance is loaded and is 0 which we also need. They do overlap in case where the balance is not loaded yet (both undefined) but its fine since then it will return on the first condition

alfetopito commented:
One other comment regarding the approval.

Nice that now I can approve without having the token!

Just that while approving COW on rinkeby, the button had no feedback that it was in progress until after it was mined
![Screen Shot 2022-07-18 at 11 01 46](https://user-images.githubusercontent.com/43217/179489687-c0b7ac3d-8378-41c0-a15e-1856b874d44e.png)

nenadV91 commented on src/custom/components/Tokens/TokensTableRow.tsx:
```diff
@@ -100,33 +101,32 @@ const DataRow = ({
+          <TableButton onClick={handleApprove} color={theme.primary1}>
             Approve all
           </TableButton>
           <ApproveLabel color={theme.green1} title={`Approved: ${currentAllowance.toExact()}`}>
```
Good catch :+1:

nenadV91 commented on src/custom/pages/Account/Tokens/TokensOverview.tsx:
```diff
@@ -1,60 +1,146 @@
-            </MainText>
-          )}
-        </AccountCard>
+          <MenuWrapper ref={node as any}>
```
Its the pattern Uni used in multiple places when the `useOnClickOutside` is used, for example https://github.com/cowprotocol/cowswap/blob/50596ae4ecdca849bce2ed832c68756b5b91df25/src/components/Settings/index.tsx#L140

alfetopito replied:
Fair enough 👍

nenadV91 commented:
@alfetopito Good catch on approve spinner, I changed some logic so this might caused the issue. I think it should be fixed now.
